src/tutorials/0006/05.vue

```diff
@@ -0,0 +1,42 @@
+<template>
+    <div>
+        <MultipleChoiceLesson
+        :text="text"
+        :question="question"
+        :choices="choices" />
+    </div>
+</template>
+
+<script>
+import MultipleChoiceLesson from '../../components/MultipleChoiceLesson'
+import text from './05.md'
+
+const question = 'How can we know the base encoding of the hash?'
+
+const choices = [
+  {
+    answer: 'We have to guess it, but it\'s easy because most implementations of IPFS encode CIDs with "base32."',
+    correct: false,
+    feedback: 'That doesn\'t sound right. Please review this lesson again.'
+  },
+  {
+    answer: 'We can check the initial characters of the hash. If it starts with "Qm", it\'s "base58btc". If it starts with "b" it\'s "base32". In both cases the CID version is "CIDv1"',
+    correct: false,
+    feedback: 'Hmm, not always. Please review this lesson again.'
+  },
+  {
+    answer: 'We can check the initial characters of the hash. If it starts with "Qm", it\'s "base58btc" and is "CIDv0." If it starts with "b" it\'s "base32" and is "CIDv1."',
```

Comment:
Not quite correct. If we have a string and we think it's a CID then we check the first character to see if it's in the mutibase table. If it's in there then we're good, provided decoding succeeds. If it's not in the table then we just _try_ to decode it using `base58btc` because that's the only other thing it could be. So there's no checking of "Qm".

> If it starts with "Qm", it\'s "base58btc"

If it starts with "Qm" it's _not_ multibase encoded, but probably a multihash encoded using base58btc.

If it starts with "z" it's probably a base58btc multibase encoded CID.

Obviously it would be awkward for us if we ever defined an encoding in multibase of "Q".

Reply:
I see. I followed the documentation [https://github.com/multiformats/cid#decoding-algorithm](here) for this bit.
I also checked the implementations, and it seems that [go-cid](https://github.com/ipfs/go-cid/blob/master/cid.go#L236) and [rust-cid](https://github.com/multiformats/rust-cid/blob/master/src/version.rs#L23) follow the [decoding algorithm spec](https://github.com/multiformats/cid#decoding-algorithm) and that [js-cid](https://github.com/multiformats/js-cid/blob/master/src/index.js#L76) and [py-cid](https://github.com/ipld/py-cid/blob/master/cid/cid.py#L266) follow the algorithm that you described if I'm not mistaken.

So which one should we consider the correct one?

Reply:
The spec - good detective work!

Reply:
I opened an issue to get this fixed up https://github.com/multiformats/js-cid/issues/103